Improve email address validation

issyl0 opened the pull request:
- We were previously only validating that the email address contains the
  `@` character. That's not enough, as people can enter invalid email
  addresses, like ones with commas in, if they mistype `.com` or
  somesuch.
- Our downstream data consumers were finding this out the hard way when
  they tried to use the email addresses to contact people.
- This uses the Python implementation at https://emailregex.com/ to
  validate the email address doesn't contain any invalid characters.

https://trello.com/c/hGSi37CH/315-email-address-field-accepts-commas

issyl0 committed:
Improve email address validation

- We were previously only validating that the email address contains the
  `@` character. That's not enough, as people can enter invalid email
  addresses, like ones with commas in, if they mistype `.com` or
  somesuch.
- Our downstream data consumers were finding this out the hard way when
  they tried to use the email addresses to contact people.
- This uses the Python implementation at https://emailregex.com/ to
  validate the email address doesn't contain any invalid characters.

https://trello.com/c/hGSi37CH/315-email-address-field-accepts-commas

bilbof reviewed (changes requested):
If possible could we have some more tests of this helper, with various email combinations? Just want to make sure we have various examples of invalid/valid emails in the test cases.

issyl0 committed:
field_validation_helper_spec: Change `context` to `describe`

- I was always taught that `context` is "when something", not "what
  something".

bilbof reviewed (approved):
🚀 awesome, thank you so much

issyl0 commented:
Thanks for the review, Bill. We can't merge this until we've heard back from DWP.

leenagupte reviewed (dismissed):
Needs stakeholder approval.

huwd commented:
@issyl0 [Steve has confirmed ](https://trello.com/c/hGSi37CH/315-email-address-field-accepts-commas)this can be merged without waiting for a reply

So looks like we can resolve that conflict and :rocket:
